lib/Tooling/CrossTranslationUnit.cpp

```diff
@@ -183,13 +183,19 @@ const FunctionDecl *CrossTranslationUnit::getCrossTUDefinition(
                      << SourceFileName << "\n";
       }
 
+      const auto& TripleTo = Context.getTargetInfo().getTriple();
+      const auto& TripleFrom = Unit->getASTContext().getTargetInfo().getTriple();
       // The imported AST had been generated for a different target
-      if (Context.getTargetInfo().getTriple() !=
-          Unit->getASTContext().getTargetInfo().getTriple()) {
+      // TODO use equality operator. Note, for some unknown reason when we do
+      // in-memory/on-the-fly CTU (i.e when the compilation db is given) some
+      // parts of the triple in the loaded ASTContext can be unknown while the
+      // very same parts in the target ASTContext are known. Thus we check for
+      // the known parts only.
+      if (!TripleTo.hasEqualKnownFields(TripleFrom)) {
```

Comment:
This function doesn't exist, thus the code doesn't compile.

Reply:
Yeah :(
Actually Triple.h is in LLVM repo, we have to fork that into Ericsson as well :(

Reply:
Looks like I can access the private variables with accessors, so we can create the function in the Clang repo, working on it.

Reply:
no, I think we should not patch LLVM just for this reason. Please implement this function into Clang.
I think we should revert this PR https://github.com/Ericsson/clang/pull/226

until this is properly solved. This blocks us now.